disaster-recovery-builder/src/main/scala/uk/gov/nationalarchives/builder/Ocfl.scala

```diff
@@ -45,27 +46,49 @@ object Ocfl:
       def loadMetadataXml(file: OcflObjectVersionFile) = XML.loadString(Files.readString(Paths.get(config.ocflRepoDir, file.getStorageRelativePath)))
 
       val ioMetadata = files.find(_.getPath.contains(s"IO_Metadata.xml")).map(loadMetadataXml)
-      val zref = ioMetadata
+
+      def getIdentifier(identifierName: String) = ioMetadata
         .flatMap { metadata =>
-          (metadata \ "Identifiers" \ "Identifier").toList
-            .find(i => (i \ "Type").text == "BornDigitalRef")
+          (metadata \ "Identifier").toList
+            .find(i => (i \ "Type").text == identifierName)
             .map(i => (i \ "Value").text)
         }
-        .getOrElse("")
+
+      val zref = getIdentifier("BornDigitalRef")
+      val sourceId = getIdentifier("SourceID")
+      val citation = getIdentifier("NeutralCitation")
 
       val title = ioMetadata
-        .map { metadata =>
-          (metadata \ "InformationObject" \ "Title").text
+        .flatMap { metadata =>
+          {
+            val titleText = (metadata \ "InformationObject" \ "Title").text
+            if (titleText.isEmpty) None else Option(titleText)
```

Comment:
I don't know. I don't want to throw errors for any missing data because there's always weirdness we don't account for. This is supposed to hold a copy of whatever is in Preservica. If there's no title then that's what we record. If one gets added later then we'll record that too.